homeassistant/components/apsystemsapi_local/sensor.py

```diff
@@ -0,0 +1,175 @@
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME, UnitOfEnergy, UnitOfPower
+from homeassistant.core import HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import ApSystemsDataCoordinator
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_IP_ADDRESS): cv.string,
+        vol.Optional(CONF_NAME, default="solar"): cv.string,
+    }
+)
+
+
+@dataclass(frozen=True, kw_only=True)
+class ApsystemsLocalApiSensorDescription(SensorEntityDescription):
+    """Describes AdGuard Home sensor entity."""
+
+    value_fn: Callable[[ReturnOutputData], float | None]
+
+
+SENSORS: tuple[ApsystemsLocalApiSensorDescription, ...] = (
+    ApsystemsLocalApiSensorDescription(
+        key="total_power",
+        translation_key="total_power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda c: c.p1 + c.p2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="total_power_p1",
+        translation_key="total_power_p1",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda c: c.p1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="total_power_p2",
+        translation_key="total_power_p2",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda c: c.p2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production",
+        translation_key="lifetime_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda c: c.te1 + c.te2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production_p1",
+        translation_key="lifetime_production_p1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda c: c.te1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production_p2",
+        translation_key="lifetime_production_p2",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
```

Comment:
Why is this total, but for `today_production` it is total_increasing?

Reply:
Today_production resets every night and lifetime_production never resets.